sdk/core/Azure.Core/src/Internal/RehydrationOperationOfT.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+#pragma warning disable SA1649 // File name should match first type name
+    internal class RehydrationOperation<T> : Operation<T> where T : IPersistableModel<T>
+#pragma warning restore SA1649 // File name should match first type name
+    {
+        private readonly OperationInternal<T> _operation;
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+
+        public RehydrationOperation(HttpPipeline pipeline, RehydrationToken? rehydrationToken, ClientOptions options = null)
+        {
+            if (pipeline is null)
+            {
+                throw new ArgumentNullException(nameof(pipeline));
+            }
+
+            if (rehydrationToken is null)
+            {
+                throw new ArgumentNullException(nameof(rehydrationToken));
+            }
```

Comment:
why not use `Argument.AssertNotNull`?

Reply:
Updated, did not use it in `NextLinkOperationImplementation` is because `Argument.cs` has been removed from shared code.